### PR TITLE
Use network object in tx builder options

### LINKED ISSUE #64: Use network instead of specifying tx version, chain ID and API URLs
Use a network object to specify tx version, chain ID and API URLs instead of setting them separately in the options object of transaction builders. Constants for Mainnet and Testnet should be made available to simplify transaction building.

Currently the options looks like
```javascript
export interface TokenTransferOptions {
  nonce?: BigNum;
  version?: TransactionVersion;
  chainId?: ChainID;
  anchorMode?: AnchorMode;
  memo?: string;
  postConditionMode?: PostConditionMode;
  postConditions?: PostCondition[];
  transferFeeEstimateApiUrl: string;
}
```

Using networks:
```
export interface TokenTransferOptions {
  fee?: BigNum;
  nonce?: BigNum;
  network?: StacksNetwork;
  anchorMode?: AnchorMode;
  memo?: string;
  postConditionMode?: PostConditionMode;
  postConditions?: PostCondition[];
}
```

### PR DESCRIPTION
**Pull request by yknl**
This PR adds the network options object to transaction builder functions. Networks are constants that will determine the transaction version, chain ID and API URLs. This simplifies the builder options and provides defaults. This PR also moves the `broadcast()` function out of the `StacksTransaction` class and into a separate function in the tx builders. This is so that users can use a network object to specify broadcast API URLs. This PR is based off of the fee estimate PR: #45 

### Usage:

```javascript
const network = new StacksTestnet();
const options = {
  network: network
};

const transaction = await makeSTXTokenTransfer(
  recipientAddress,
  amount,
  secretKey,
  options
);

broadcastTransaction(transaction, network);
```

### Creating a custom network
```javascript
const myNetwork = {
  version: TransactionVersion.Mainnet;
  chainId: ChainID.Mainnet;
  coreApiUrl = 'https://mystacksnetworknode.com';
  broadcastApiUrl = `${this.coreApiUrl}/v2/transactions`;
  transferFeeEstimateApiUrl = `${this.coreApiUrl}/v2/fees/transfer`;
}
```

Resolves #64

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] API reference/documentation update
- [ ] Other

## Does this introduce a breaking change?
Yes, the options object for transaction builder functions have been changed.

`TokenTransferOptions`
`ContractDeployOptions`
`ContractCallOptions`

The `StacksTransaction` class no longer has a `broadcast()` method. It has been moved to a separate function `broadcastTransaction()`. And it now takes a transaction and network as arguments.

## Are documentation updates required?
The readme examples have been updated with this PR.

## Testing information

Provide context on how tests should be performed.
Transactions should continue to be generated and broadcasted correctly when using the new options objects for transaction builders. 

## Checklist
- [x] Code is commented where needed
- [x] Unit test coverage for new or modified code paths
- [x] `npm run test` passes
- [ ] Changelog is updated
- [x] Tag 1 of @yknl, @zone117x, @reedrosenbluth for review

### CODE REVIEW

File: src/network.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { TransactionVersion, ChainID } from './constants';
+
+export interface StacksNetwork {
```
Comment (zone117x) [inline src/network.ts:3]: This is great API design, nice work @yknl

File: tests/src/builder-tests.ts [hunk 2]
```diff
@@ -376,7 +376,38 @@ test('Estimate token transfer fee', async () => {
...
+  const memo = 'test memo';
+
+  const network = new StacksMainnet();
+
+  const options = {
+    fee,
+    memo: memo,
+  };
+
+  const transaction = await makeSTXTokenTransfer(recipient, amount, secretKey, options);
```
Comment (zone117x) [inline tests/src/builder-tests.ts:404]: I think the general trend in highly parameterized js functions (and one I've noticed at Blockstack), is that we start off with a few reasonable non-optional args and a couple reasonable optional args, then eventually end up with a giant list of top-level optional args. 

I propose we take this breaking change as an opportunity to make all of these building functions take an object (as the only top-level arg), and control the optional and non-optional (now and in the future) args using the cleaner named object param.

Comment (hstove) [reply tests/src/builder-tests.ts:404]: Yes, I would strongly 👍 this.

File: src/network.ts (hunk 1 above)
Comment (agraebe) [reply src/network.ts:3]: 💯

File: tests/src/builder-tests.ts (hunk 2 above)
Comment (yknl) [reply tests/src/builder-tests.ts:404]: Great suggestion 👍 We will address this in a separate PR.

**Pull request merged**